parts/kubernetesagentcustomdata.yml

```diff
@@ -128,7 +128,8 @@ write_files:
   owner: "root"
   content: |
     #!/bin/bash
-    exit 0
+    echo $(curl -q http://169.254.169.254/metadata/v1/InstanceInfo 2>/dev/null | sed "s/.*\"FD\":\"\([0-9]*\).*/\1/") > /etc/kubernetes/fd
+    echo "',variables('{{.Name}}VMSize'),'" > /etc/kubernetes/vmsize
```

Comment:
vmsize is a Azure concept. Not sure if this should be located at /etc/kubernetes/ hierarch.